samcli/lib/providers/sam_function_provider.py

```diff
@@ -235,12 +265,52 @@ def _convert_sam_function_resource(
             LOG.debug("Found Serverless function with name='%s' and ImageUri='%s'", name, imageuri)
 
         return SamFunctionProvider._build_function_configuration(
-            stack, name, codeuri, resource_properties, layers, inlinecode, imageuri, use_raw_codeuri
+            stack, function_id, name, codeuri, resource_properties, layers, inlinecode, imageuri, use_raw_codeuri
         )
 
+    @staticmethod
+    def _get_function_id(resource_properties: Dict, logical_id: str) -> str:
+        """
+        Get unique id for Function resource.
+        For CFN/SAM project, this function id is the logical id.
+        For CDK project, this function id is the user-defined resource id, or the logical id if the resource id is not
+        found.
+
+        Parameters
+        ----------
+        resource_properties str
+            Properties of this resource
+        logical_id str
+            LogicalID of the resource
+
+        Returns
+        -------
+        str
+            The unique function id
+        """
+        resource_cdk_path = resource_properties.get("Metadata", {}).get("aws:cdk:path")
+
+        if not isinstance(resource_cdk_path, str) or not resource_cdk_path:
+            return logical_id
+
+        # aws:cdk:path metadata format: {stack_id}/{function_id}/Resource
```

Comment:
Are we sure this is the only format to `aws:cdk:path`? Maybe link out to docs?

Reply:
Sure. I have added the cdk document about this in the comment.